### DISCUSSION TITLE
Api Server Integration

### DISCUSSION THREAD

**Discussion by MichalJef** [pear-devs/pear-desktop#2765] [Q&A] [answered]
So I found out that there is a plugin _API Server[Beta]_. It's in Beta, so maybe that's why I'm not able to work with it. However, I think the problem is me. When I get to the IP and PORT I get a `404 Not Found` error, which is atypical for a browser. So I guess I successfully connect to the API but I don't know what to do next, I've tried different endpoints like `/play`; `/pause`; /`status` or none (if this is even way it works).
I tried exactly: `http://localhost:port` (in my case port `8080`).
I'm asking because I haven't found any documentation, but given the _[New]_ tag I understand I haven't found anything. That's why I'm wondering if this is in the works and if or how to connect to it. And yes, i know about `youtubemusic://[function]` but that's quite different.

Has anyone managed to get it working? Or does anyone know how to use it?

EDIT:
I found files with this API. However, it didn't solve my problem. I would be glad for some advice.

**Comment by JellyBrick** [comment, marked as answer]
Did you try `http://localhost:port/swagger`?

**Reply by MichalJef** [reply]
Thanks, that's what I wanted.

I just wanna ask, if I can somehow set up CORS on the server.

**Reply by MichalJef** [reply]
Okay so the final solution with CORS is to use IP `0.0.0.0` which will make it available for use by all devices on the network, instead of `127.0.0.1`.
You need to use CURL (in HTML as JS).'

### Example:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>API Test</title>
</head>
<body>
    <button id="togglePlay">Toggle Play</button>

    <script>
        document.getElementById('togglePlay').addEventListener('click', () => {
            fetch('http://localhost:port/api/v1/toggle-play', {
                method: 'POST',
                headers: {
                    'Accept': '*/*'
                }
            })
            .then(response => {
                if (response.ok) {
                    return response.text();
                } else {
                    throw new Error(`HTTP error! status: ${response.status}`);
                }
            })
            .then(data => {
                console.log('Response:', data);
            })
            .catch(error => {
                console.error('Error:', error);
            });
        });
    </script>
</body>
</html>
```
A solution for anyone who finds themselves in a situation like mine.
